cmd/main.go

```diff
@@ -704,19 +704,13 @@ func createS3Bucket(objectHandler s3wrapper.API, log logrus.FieldLogger) {
 
 func createStorageClient(deployTarget string, storage string, s3cfg *s3wrapper.Config, fsWorkDir string,
 	log logrus.FieldLogger, metricsAPI metrics.API, fsThreshold int) s3wrapper.API {
-	var storageClient s3wrapper.API
+	var storageClient s3wrapper.API = nil
 	if storage != "" {
 		switch storage {
 		case storage_s3:
 			storageClient = s3wrapper.NewS3Client(s3cfg, log)
-			if storageClient == nil {
```

Comment:
can you explain these changes?

Reply:
Sure.
`golangci-lint` noted that those `if` statements are redundant, but now when looking on those statements again, only `s3wrapper.NewFSClient` function will never be nil.
```
cmd/main.go:711:58: SA4023: this comparison is never true (staticcheck)
                        if storageClient = s3wrapper.NewS3Client(s3cfg, log); storageClient == nil {
                                                                              ^
cmd/main.go:715:87: SA4023: this comparison is never true (staticcheck)
                        if storageClient = s3wrapper.NewFSClient(fsWorkDir, log, metricsAPI, fsThreshold); storageClient == nil {
                                                                                                           ^
cmd/main.go:725:58: SA4023: this comparison is never true (staticcheck)
                        if storageClient = s3wrapper.NewS3Client(s3cfg, log); storageClient == nil {
```

Reply:
https://github.com/openshift/assisted-service/blob/master/pkg/s3wrapper/filesystem.go#L32

Reply:
I'll push fix for that